service/controller/resource/azureconfig/ingress.go

```diff
@@ -0,0 +1,50 @@
+package azureconfig
+
+import (
+	"strings"
+
+	"github.com/giantswarm/microerror"
+	capzv1alpha3 "sigs.k8s.io/cluster-api-provider-azure/api/v1alpha3"
+)
+
+const (
+	ingressControllerDomainPrefix = "ingress"
+	ingressWildcardDomainPrefix   = "*"
+)
+
+func (r *Resource) newIngressDomain(cr capzv1alpha3.AzureCluster, domainPrefix string) (string, error) {
```

Comment:
Just by reading their names, it's hard to tell which one is which. With a sentence explaining what they do we don't have to go through the code. I believe it would benefitial to add docs to these functions.